enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -352,20 +346,31 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
         if not self.catalog_contains_content_key(content_key):
             return (False, REASON_CONTENT_NOT_IN_CATALOG, [])
 
+        # Wait to fetch content metadata with a call to the enterprise-subsidy
+        # service until we *know* that we'll need it.
+        content_metadata = self.get_content_metadata(content_key)
+
         # no content key in content metadata
         if not content_metadata:
-            return (False, REASON_CONTENT_NOT_IN_CATALOG, existing_transactions)
+            return (False, REASON_CONTENT_NOT_IN_CATALOG, [])
 
         # TODO: Add Course Upgrade/Registration Deadline Passed Error here
 
-        # inactive subsidy
+        # We want to wait to do these checks that might require a call
+        # to the enterprise-subsidy service until we *know* we'll need the data.
+        subsidy_can_redeem_payload = self.subsidy_client.can_redeem(
+            self.subsidy_uuid,
+            lms_user_id,
+            content_key,
+        )
+
+        active_subsidy = subsidy_can_redeem_payload.get('active', False)
```

Comment:
[suggestion/nit] Consider adding a code comment that this `active` refers to a computed property that takes into account start/expiration dates so there's less confusion between the subsidy `active` value versus the policy's `is_active` flag?